pkg/controller/clusterdeployment/clusterdeployment_deleted.go

```diff
@@ -42,27 +42,32 @@ func (r *ReconcileClusterDeployment) handleDelete(request reconcile.Request, ins
 		BaseDomain: instance.Spec.BaseDomain,
 	}
 	err := pdData.ParsePDConfig(r.client)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
+	// if there is an error, no configuration (configmap or secret) was found.
+	// we are deleting.
+	// if we block deletion because of missing config we will block cluster deletion.
+	// if we do not block we could leave a dangling service in PD.
+	// without the config we cannot fix the dangling service.
+	// THEREFORE: do not fail if we can't find PD config, just continue
+	if err == nil {
+		err = pdData.ParseClusterConfig(r.client, request.Namespace, request.Name)
 
-	err = pdData.ParseClusterConfig(r.client, request.Namespace, request.Name)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
+		if err == nil {
```

Comment:
same here.. shouldn't we parse a specific error condition?

Reply:
+1